components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2Test.java

```diff
@@ -121,4 +138,111 @@ public void testPatchRoleWithGroupDisplayNameInsteadOfGroupIdThrowingErrors(Stri
+
+            List<String> requiredAttributes = new ArrayList<>();
+            requiredAttributes.add("properties");
+
+            Role mockedRole1 = new Role();
+            mockedRole1.setId(ROLE_ID);
+            mockedRole1.setName(ROLE_NAME);
+            mockedRole1.setAudience(ORGANIZATION_AUD);
+            mockedRole1.setAudienceId(ORGANIZATION_ID);
+            mockedRole1.setAudienceName(ORGANIZATION_NAME);
+
+            Role mockedRole2 = new Role();
+            mockedRole2.setId("role_id_2");
+            mockedRole2.setName("role_name_2");
+            mockedRole2.setAudience(ORGANIZATION_AUD);
+            mockedRole2.setAudienceId(ORGANIZATION_ID);
+            mockedRole2.setAudienceName(ORGANIZATION_NAME);
+
+            if (isPropertiesRequired) {
+                RoleProperty roleProperty1 = new RoleProperty();
+                roleProperty1.setName(ROLE_PROPERTY_NAME);
+                roleProperty1.setValue(Boolean.TRUE.toString());
+                mockedRole1.setRoleProperty(roleProperty1);
+
+                RoleProperty roleProperty2 = new RoleProperty();
+                roleProperty2.setName(ROLE_PROPERTY_NAME);
+                roleProperty2.setValue(Boolean.FALSE.toString());
+                mockedRole2.setRoleProperty(roleProperty2);
+            }
+
+            List<Role> mockedRoles = new ArrayList<>();
+            mockedRoles.add(mockedRole1);
+            mockedRoles.add(mockedRole2);
+
+            mockedSCIMCommonUtils.when(() -> SCIMCommonUtils.getSCIMRoleV2URL(ROLE_ID)).
+                    thenReturn(SCIM2_ROLES_V2_LOCATION_URI_BASE + ROLE_ID);
+            mockedSCIMCommonUtils.when(() -> SCIMCommonUtils.getSCIMRoleV2URL(ROLE_ID_2)).
+                    thenReturn(SCIM2_ROLES_V2_LOCATION_URI_BASE + ROLE_ID_2);
+
+            when(roleManagementService.getRoles(10, 1, null, null, SAMPLE_TENANT_DOMAIN, requiredAttributes)).
+                    thenReturn(mockedRoles);
+
+            RolesV2GetResponse rolesV2GetResponse = scimRoleManagerV2.listRolesWithGET(null, 1, 10, null, null,
+                    requiredAttributes);
+            List<RoleV2> roles = rolesV2GetResponse.getRoles();
+            assertEquals(roles.get(0).getDisplayName(), ROLE_NAME);
+            assertEquals(roles.get(0).getLocation(), SCIM2_ROLES_V2_LOCATION_URI_BASE + ROLE_ID);
+
+            assertEquals(roles.get(1).getDisplayName(), ROLE_NAME_2);
+            assertEquals(roles.get(1).getLocation(), SCIM2_ROLES_V2_LOCATION_URI_BASE + ROLE_ID_2);
+
+            if (isPropertiesRequired) {
+                assertEquals(roles.get(0).getRoleProperties().get(0).getDisplay(), ROLE_PROPERTY_NAME);
+                assertEquals(roles.get(0).getRoleProperties().get(0).getValue(), Boolean.TRUE.toString());
+                assertEquals(roles.get(1).getRoleProperties().get(0).getDisplay(), ROLE_PROPERTY_NAME);
+                assertEquals(roles.get(1).getRoleProperties().get(0).getValue(), Boolean.FALSE.toString());
+            }
```

Comment:
else, assert that properties are not turned

Reply:
Changed